api/appointmentsPOST.ts

```diff
@@ -47,40 +49,55 @@ let appointment_datapoints : string [] = [
   'audiologistexamsid',
   'appointmentdatetime'
 ]
+let appointment_datapoints_MAP : string [] = [
+  'authorityid',
+  'patientid',
+  'tfisurveyid',
+  'thssurveyid',
+  'tssurveyid',
+  'audiologistexamsid',
+]
 
-export default handler(async (request: any) => {
+export default handler(async (request: any, userId) => {
   
+  let requestVars:AppointmentSubmission =  new AppointmentSubmission(request.body);
+  let body = requestVars.exportToSQLVars();
+
   return await db(async (connection) => {
     // Take data from request and build sql call from i
   let tfisurvey_sql : string = "INSERT INTO tfisurvey (" + tfi_datapoints.join() + ") VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) RETURNING *"; 
-  let tfisurvey_value : number[] = tfi_datapoints.map((value, index) => request.body[value])
+  let tfisurvey_value : number[] = tfi_datapoints.map((value) => body[value])
 
   let thssurvey_sql : string = "INSERT INTO thssurvey (" + ths_datapoints.join() + ") VALUES ($1, $2, $3, $4, $5) RETURNING *";
-  let thssurvey_values = ths_datapoints.map((value, index) => request.body[value])
+  let thssurvey_values = ths_datapoints.map((value) => body[value])
 
   let tssurvey_sql : string = "INSERT INTO tssurvey (" + ts_datapoints.join() + ") VALUES ($1, $2) RETURNING *";
-  let tssurvey_values = ts_datapoints.map((value, index) => request.body[value])
+  let tssurvey_values = ts_datapoints.map((value) => body[value])
 
   let audiologistexam_sql : string = "INSERT INTO audiologistexams (" + audiologistexams_datapoints.join() + ") VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING *";
-  let audiologistexam_values = audiologistexams_datapoints.map((value, index) => request.body[value])
+  let audiologistexam_values = audiologistexams_datapoints.map((value) => body[value]);
 
   // Appointments inserted last because it needs info from the other tables
-    await Promise.all([
+    return await Promise.all([
```

Comment:
use `async` style by replacing this with `const values = await Promise....`

Reply:
As in rewrite this line to 
`const values = await Promise.all([`
then add a line after the catch:
`return values;`
Right?

Reply:
No. `async` style eliminates the call to `then`. What makes `async` functions easier to read is that they are not written with calls to `then` (even though they are secretly implemented with them). 

When you write `const foo = await bar; const foo1 = foo + 1;`, this is translated to `bar.then(foo => { const foo1 = foo + 1; }`

`values` here is the same variable as (and eliminates) the function parameter `values` to the lambda. Eliminate the usage of this lambda.